lib/OpenQA/Worker.pm

```diff
@@ -104,6 +107,32 @@ sub catch_exit {
     }
 }
 
+sub setup_logging {
+    my ($app) = @_;
+
+    my $level = $worker_settings->{LOG_LEVEL} ? $worker_settings->{LOG_LEVEL} : "debug";
+
+    if (my $logfolder = $ENV{OPENQA_WORKER_LOG} || $worker_settings->{LOG_FOLDER}) {
+        # So each worker from each host get it's own log (as the folder can be shared). Hopefully the machine hostname
+        # is already sanitized. Otherwise we need to check
+        $app->log->path(catfile($logfolder, hostname() . "-$instance.log"));
+        $app->log->format(
+            sub {
+                my ($time, $level, @lines) = @_;
+                return '[' . localtime($time) . "] [" . $app->log_name . ":$level] " . join "\n", @lines, '';
+            });
+
+    }
+    elsif (lc($level) eq 'debug' || lc($level) eq 'info') {
+        $app->log->handle(\*STDOUT);
+    }
+
+    $app->log->level($level);
```

Comment:
same here, better declare a default level when contructing the `Mojo::Log` object, and edit later if another level is specified in the environment variable or in config file